firedrake/slate/slate.py

```diff
@@ -620,3 +607,21 @@ def __repr__(self):
     def _key(self):
         """Returns a key for hash and equality."""
         return (type(self), self.operands, self.actee)
+
+
+# Establishes levels of precedence for Slate tensors
+precedences = [
+    [Tensor],
+    [UnaryOp],
+    [Add, Sub],
+    [Mul, Action]
+]
+
+# Here we establish the precedence class attribute for a given
+# Slate tensor. Unary and binary operations will have higher
+# precedence over terminal Tensor objects. Similarly, tensor
+# action and multiplication will have higher precedence over
+# addition/subtraction
```

Comment:
The first sentence of this comment is fine, but the rest is at best misplaced. The following code does not authoritatively define precedence, it just assigns the levels to the classes. The definition happens above. Moreover, one must always be cautious when wanting to repeat in comment exactly what the code does: it often does not help much, but there is danger that when the code is changed in the future, someone will forget to update the comments as well.